Fix memory leak, issue-59

Kdoggett887 opened the pull request:
Resets a class variable after each extraction to prevent memory leak. More details in #59 .

lopuhin commented:
@Kdoggett887 thanks for PR! Sorry if the answer to my question is obvious, as I'm new to extruct, but I wonder why is ``_lookups`` a class attribute instead of an instance attribute? As far as I understand, if it would be an instance attribute, then it would be cleaned up automatically since we create a new parser in ``extruct.extract`` function?

Kdoggett887 commented:
@lopuhin Thanks for the suggestion! I was looking into doing this, but was unsuccessful at first and gave up on it fairly quickly since it caused some upstream issues in the lxml package which I'm unfamiliar with.

After your suggestion, I decided to look into it more and found a solution that puts the `_lookups` variable into the constructor. When I had tried it before, I wasn't instantiating the `HtmlElementClassLookup`, so the constructor just instantiates the super class and assigns `self._lookups = {}`. I'm testing the memory usage on a large sample of files now, but if it works this should be a better solution.

kmike commented on tests/test_extruct.py:
```diff
@@ -3,6 +3,7 @@
 import unittest
 
 import extruct
+from extruct.xmldom import DomHtmlElementClassLookup
```
why is this import needed?

Kdoggett887 commented:
@lopuhin thanks for the feedback! I fixed up the spacing and removed the unused import. I ran a test checking the speed and the updated version took ~85% of the time master did, but more testing would be needed to confirm the results.

kmike commented:
Awesome, thanks @Kdoggett887!
